docs/Release-Process.md

```diff
@@ -5,44 +5,69 @@ Containers release
+the plan is to do a succession of component releases, rather than releasing
+all components at the same time. This means that the peer pods release process needs to slot into the correct place.
+The flow of releases should roughly be:
+1. The [guest-components release](https://github.com/confidential-containers/guest-components/releases) (or a pinned
+version is picked) and [kata-containers](https://github.com/kata-containers/kata-containers) bumps these versions in
+    [versions.yaml](https://github.com/kata-containers/kata-containers/blob/main/versions.yaml) under
+    `externals.coco-guest-components.version` and the `image-rs` crate in the agent's
+    [`Cargo.toml`](https://github.com/kata-containers/kata-containers/blob/main/src/agent/Cargo.toml).
+    - At this point it makes sense for us to try and bump the guest-components that are built into the peer pod vm image,
+    set by the `git.guest-components.reference` value in [versions.yaml](../src/cloud-api-adaptor/versions.yaml)
+    to match the released versions. For compatibility, we might also need to bump the kata-containers agent to the latest commit
+    hash in our [version.yaml](../src/cloud-api-adaptor/versions.yaml) for testing.
+1. Kata Containers [releases](https://github.com/kata-containers/kata-containers/releases)
+    - Hopefully we've already bumped the guest-components in the previous step, but now we should update:
+      - The kata-containers source branch that we use in [versions.yaml](../src/cloud-api-adaptor/versions.yaml) to
 the tag of the kata-containers release candidate.
-- The `kata-containers/src/runtime` go module that we include in the main `cloud-api-adaptor` [`go.mod`](../src/cloud-api-adaptor/go.mod) and the `csi-wrapper` [`go.mod`](../src/csi-wrapper/go.mod).
-This can be done by running
-    ```
-    go get github.com/kata-containers/kata-containers/src/runtime@<release candidate branch e.g. CCv0>
-    go mod tidy
-    ```
-in the [cloud-api-adaptor](../src/cloud-api-adaptor/) directory and [csi-wrapper](../src/csi-wrapper/) directory.
-
-- The attestation-agent that is built into the peer pod vm image, by updating the `git.guest-components.reference` value in [versions.yaml](../src/cloud-api-adaptor/versions.yaml)
+      - The `kata-containers/src/runtime` go module that we include in the main `cloud-api-adaptor` [`go.mod`](../src/cloud-api-adaptor/go.mod) and the `csi-wrapper` [`go.mod`](../src/csi-wrapper/go.mod). This can be done by running
+        ```
+        go get github.com/kata-containers/kata-containers/src/runtime@<latest release e.g. 3.6.0>
+        go mod tidy
+        ```
+        in the [cloud-api-adaptor](../src/cloud-api-adaptor/) directory and [csi-wrapper](../src/csi-wrapper/) directory.
+1. Trustee releases?
+    - We should bump the `kbs` version in [versions.yaml](../src/cloud-api-adaptor/versions.yaml)
+     to match the released and tested version of trustee, if applicable
```

Comment:
This should probably be coupled with the releases of guest-components. i think we will almost always tag those together.

Reply:
Thanks for the info. I've updated the doc in a separate commit to make it easier to re-check, but will fixup before this gets merged.